hyperspy/_signals/eds.py

```diff
@@ -618,3 +623,187 @@ def get_take_off_angle(self):
+            If False, use all the lines of each element in the data spectral
+            range. If True use only the line at the highest energy
+            above an overvoltage of 2 (< beam energy / 2).
+        integration_windows: None or float or 2D array of float
+            If not None, add markers at the position of the integration
+            windows.
+            If float, use 'estimate_integration_windows`.
+            Else provide an array for which each row corresponds to a X-ray
+            line. Each row contains the left and right value of the window.
+        kwargs
+            The extra keyword arguments for plot()
+
+        Examples
+        -------
+
+        >>> s.plot(True, xray_lines=['Mn_Ka'], integration_windows='auto')
+
+        See also
+        --------
+        set_elements, add_elements, estimate_integration_windows
+
+        """
+        super(EDSSpectrum, self).plot(**kwargs)
+        if xray_lines_markers:
+            if only_lines is not None:
+                only_lines = list(only_lines)
+                for only_line in only_lines:
+                    if only_line == 'a':
+                        only_lines.extend(['Ka', 'La', 'Ma'])
+                    elif only_line == 'b':
+                        only_lines.extend(['Kb', 'Lb1', 'Mb'])
+            if xray_lines is None or xray_lines == 'from_elements':
+                if 'Sample.xray_lines' in self.metadata \
+                        and xray_lines != 'from_elements':
+                    xray_lines = self.metadata.Sample.xray_lines
+                elif 'Sample.elements' in self.metadata:
+                    xray_lines = self._get_lines_from_elements(
+                        self.metadata.Sample.elements,
+                        only_one=only_one,
+                        only_lines=only_lines)
+                else:
+                    raise ValueError(
+                        "No elements defined, set them with `add_elements`")
+            xray_lines, xray_not_here = self._get_xray_lines_in_spectral_range(
+                xray_lines)
+            for xray in xray_not_here:
+                print("Warning: %s is not in the data energy range." % (xray))
+            xray_lines = np.unique(xray_lines)
+            self._add_xray_lines_markers(xray_lines)
+            if integration_windows is not None:
+                if integration_windows == 'auto':
+                    integration_windows = self.estimate_integration_windows(
+                        xray_lines=xray_lines)
+                self._add_vertical_lines(integration_windows, linestyle='--')
+
+    def _add_vertical_lines(self, position, **kwargs):
+        """
+        Add vertical markers for each X-ray lines.
+
+        Parameters
+        ----------
+        position: 2D array of float
+            The position of the lines in energy. Each row corresponds to a
+            X-ray lines.
+        kwargs
+            keywords argument for markers.vertical_line
+        """
```

Comment:
As per our discussion, maybe rename to `_add_vertical_line_groups`? 

In the documentation, maybe changing "X-ray" to "group (of)" is sufficient?